### PR TITLE
mrc-706 delete files front-end

### PR DESCRIPTION
**Pull request by hillalex**
Preliminary PR with the actions and integration tests. Not wired up to the UI yet.

Quite a lot of lines of code changed but it's very boilerplate-y. Was wondering if we should make things less boilerplate-y by having single upload/delete methods that take the filetype as a param instead? I guess there's some trade-off with clarity vs repetitiveness.

### CODE REVIEW

File: src/app/static/src/app/store/baseline/mutations.ts [hunk 1]
```diff
@@ -35,6 +35,8 @@ export const mutations: MutationTree<BaselineState> & BaselineMutations = {
     PJNZUpdated(state: BaselineState, action: PayloadWithType<PjnzResponse | null>) {
         const data = action.payload;
         if (data) {
+            console.log("----------------")
```
Comment (EmmaLRussell) [inline src/app/static/src/app/store/baseline/mutations.ts]: You want this to stay in or was it just for testing?

File: src/app/static/src/app/store/surveyAndProgram/actions.ts [hunk 2]
```diff
@@ -62,6 +66,41 @@ export const actions: ActionTree<SurveyAndProgramDataState, RootState> & SurveyA
             });
     },
 
+    async deleteSurvey({commit}) {
+        await api(commit)
+            .delete("/disease/survey/")
+            .then(() => {
+                commit({type: "SurveyUpdated", payload: null});
+                commit({type: "filteredData/Reset", payload: null}, {root: true});
```
Comment (EmmaLRussell) [inline src/app/static/src/app/store/surveyAndProgram/actions.ts]: Should this be resetting the other module steps too (modelRunResponse etc)? Or will this be done in a separate ticket? Feels like we should maybe have a utility resetStateFrom method which will reset all models subsequent to a given module (and deal with the fact that some modules aren't explicitly associated with a step e.g. filteredData)

Comment (hillalex) [reply src/app/static/src/app/store/surveyAndProgram/actions.ts]: Yes this becomes obsolete in the next ticket https://vimc.myjetbrains.com/youtrack/issue/mrc-586

File: src/app/static/src/app/store/baseline/mutations.ts (hunk 1 above)
Comment (hillalex) [reply src/app/static/src/app/store/baseline/mutations.ts]: oops :) removed

File: src/app/static/src/app/store/baseline/actions.ts [hunk 3]
```diff
@@ -13,9 +13,13 @@ export type BaselineErrorActionTypes =
 
 export interface BaselineActions {
     uploadPJNZ: (store: ActionContext<BaselineState, RootState>, formData: FormData) => void
-    getBaselineData: (store: ActionContext<BaselineState, RootState>) => void
     uploadShape: (store: ActionContext<BaselineState, RootState>, formData: FormData) => void,
     uploadPopulation: (store: ActionContext<BaselineState, RootState>, formData: FormData) => void,
+    deletePJNZ: (store: ActionContext<BaselineState, RootState>) => void
+    deleteShape: (store: ActionContext<BaselineState, RootState>) => void,
+    deleteAll: (store: ActionContext<BaselineState, RootState>) => void,
```
[the remark this answers is not in the document]
Comment (hillalex) [reply src/app/static/src/app/store/baseline/actions.ts]: If the app is loaded in an invalid state we wipe all inputs to start from fresh: https://github.com/mrc-ide/hint/pull/158/files#diff-10c3cdc0d39b76f3fd6622aed354ef5fR21

File: src/app/static/src/app/store/surveyAndProgram/actions.ts (hunk 2 above)
Comment (EmmaLRussell) [reply src/app/static/src/app/store/surveyAndProgram/actions.ts]: Ha, yup, I completely forgot what the whole point of that conversation was...

**Pull request merged**